src/Discord/Parts/Guild/Guild.php

```diff
@@ -655,6 +657,53 @@ public function searchMembers(array $options): ExtendedPromiseInterface
         });
     }
 
+    /**
+     * Returns the Welcome Screen object for the guild.
+     *
+     * @return ExtendedPromiseInterface
+     */
+    public function getWelcomeScreen(): ExtendedPromiseInterface
```

Comment:
Since the welcome screen is only present when sent down inside an invite, maybe we should 'hide' the `welcome_screen` attribute and force the users to use `getWelcomeScreen()`.

e.g. remove `welcome_screen` from the phpdoc (but leave it in fillable), then in this function check `isset($this->attributes['welcome_screen'])` and return that if so

Reply:
The `welcome_screen` may be updated,  it's not only present with invites.

I think to make it like the fetch() function:
```suggestion
    public function getWelcomeScreen(bool $fresh = false): ExtendedPromiseInterface
    {
        if (! $fresh && isset($this->attributes['welcome_screen'])) {
            return \React\Promise\resolve($this->attributes['welcome_screen']);
        }

```